controllers/humiocluster_controller.go

```diff
@@ -2349,3 +2426,184 @@ func getHumioNodePoolManagers(hc *humiov1alpha1.HumioCluster) HumioNodePoolList
+	pdbName := hnp.GetPodDisruptionBudgetName() // Use GetPodDisruptionBudgetName from HumioNodePool
+	// Create and populate labels map
+	labels := kubernetes.LabelsForHumio(hc.Name)
+	labels["humio.com/node-pool"] = hnp.GetNodePoolName()
+
+	selector := &metav1.LabelSelector{
+		MatchLabels: kubernetes.MatchingLabelsForHumio(hc.Name), // Assuming PDB should target all Humio pods
+	}
+
+	minAvailable := pdbSpec.MinAvailable
+	pdb := &policyv1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       pdbName,
+			Namespace:  hc.Namespace,
+			Labels:     kubernetes.LabelsForHumio(hc.Name), // Add node pool name label if needed
+			Finalizers: []string{HumioProtectionFinalizer},
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(hc, humiov1alpha1.GroupVersion.WithKind("HumioCluster")),
+			},
+		},
+		Spec: policyv1.PodDisruptionBudgetSpec{
+			Selector: selector,
+		},
+	}
+
+	if minAvailable != nil {
+		pdb.Spec.MinAvailable = minAvailable
+	} else {
+		defaultMinAvailable := intstr.IntOrString{
+			Type:   intstr.Int,
+			IntVal: 1,
+		}
+		pdb.Spec.MinAvailable = &defaultMinAvailable
+	}
+
+	return pdb, nil
+}
+
+// createOrUpdatePDB creates or updates a PodDisruptionBudget object
+func (r *HumioClusterReconciler) createOrUpdatePDB(ctx context.Context, hc *humiov1alpha1.HumioCluster, hnp *HumioNodePool, desiredPDB *policyv1.PodDisruptionBudget) error {
+	currentPDB := &policyv1.PodDisruptionBudget{}
+	pdbName := hnp.GetPodDisruptionBudgetName()
+	err := r.Get(ctx, client.ObjectKey{Name: pdbName, Namespace: hc.Namespace}, currentPDB)
+	if err != nil && k8serrors.IsNotFound(err) {
+		r.Log.Info("creating PDB", "pdb", pdbName)
+		err = r.Create(ctx, desiredPDB)
+		if err != nil {
+			r.Log.Error(err, "failed to create PDB", "pdbName", desiredPDB.Name, "pdbNamespace", desiredPDB.Namespace)
+			return fmt.Errorf("failed to create PDB %s/%s: %w", desiredPDB.Namespace, desiredPDB.Name, err)
+		}
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("failed to get PDB: %w", err)
+	}
+
+	if !SemanticPDBsEqual(desiredPDB, currentPDB) {
+		r.Log.Info("updating PDB", "pdb", pdbName)
+		updatedPDB := currentPDB.DeepCopy()
+		updatedPDB.Spec = desiredPDB.Spec
+		err = r.Update(ctx, updatedPDB)
+		if err != nil {
+			r.Log.Error(err, "failed to update PDB", "pdbName", desiredPDB.Name, "pdbNamespace", desiredPDB.Namespace)
+			return fmt.Errorf("could not update PDB %s/%s: %w", desiredPDB.Namespace, desiredPDB.Name, err) // Return Result{}, error on failure
+		}
+	}
+
+	r.Log.V(1).Info("pdb is up-to-date", "pdb", pdbName)
+	return nil
+}
+
+// SemanticPDBsEqual compares two PodDisruptionBudgets and returns true if they are equal
+func SemanticPDBsEqual(desired *policyv1.PodDisruptionBudget, current *policyv1.PodDisruptionBudget) bool {
+	if !equality.Semantic.DeepEqual(desired.Spec.MinAvailable, current.Spec.MinAvailable) {
```

Comment:
Cool. Wasn't aware of this `equality` package. 

I see we have a total of 10 references to `reflect.DeepEqual()` various places in the code base on master branch, and maybe we should consider refactoring those to use `equality.Semantic.DeepEqual()` instead.

We also have 91 places on master where we use `cmp.diff()`. Not all of those actually _need_ the diff so perhaps we could also replace and update those with `equality.Semantic.DeepEqual()`.

I don't expect this to be incorporated in this PR though, just wanted to call it out as a possible cleanup we may consider later as I wasn't aware of this `equality` package.

Reply:
II also would like to tackle those changes on a different PR. Thanks for pointing that out